src/newgrf_text.cpp

```diff
@@ -206,20 +209,17 @@ struct UnmappedChoiceList {
 			for (int i = 0; i < count; i++) {
 				int idx = (this->type == SCC_GENDER_LIST ? lm->GetReverseMapping(i, true) : i + 1);
 				const auto &str = this->strings[this->strings.find(idx) != this->strings.end() ? idx : 0].str();
-				size_t len = str.size() + 1;
-				if (len > 0xFF) GrfMsg(1, "choice list string is too long");
-				*d++ = GB(len, 0, 8);
+				size_t len = str.size();
+				if (len > UINT8_MAX) GrfMsg(1, "choice list string is too long");
+				*d++ = ClampTo<uint8_t>(len);
```

Comment:
I guess just a `static_cast` would suffice here. Or is this preparing for some further work?

Reply:
A `static_cast` would convert a 257 to 1.
The goal is to truncate the string at position 256.

Reply:
Hmm, I kinda assumed the error to be fatal. But in retrospect it's not. My bad.